src/coreclr/jit/codegenxarch.cpp

```diff
@@ -252,7 +252,8 @@ BasicBlock* CodeGen::genCallFinally(BasicBlock* block)
 #endif // JIT32_GCENCODER
 
         // Now go to where the finally funclet needs to return to.
-        if (block->bbNext->bbJumpDest == block->bbNext->bbNext)
+        if ((block->bbNext->bbJumpDest == block->bbNext->bbNext)
+            && !compiler->fgInDifferentRegions(block->bbNext, block->bbNext->bbJumpDest))
```

Comment:
Should we revise this (and the "copies" in other archs) to factor out some of these repeated indirections, eg this seems more readable to me:
```suggestion
       BasicBlock* const nextBlock = block->bbNext;
       BasicBlock* const jumpDest  = nextBlock->bbJumpDest;
        if ((jumpDest == nextBlock->bbNext)
            && !compiler->fgInDifferentRegions(nextBlock, jumpDest))
```
Likely the compiler is doing this behind the scenes anyways so this is more about readabilty than about perf.

Reply:
Also note there are uses of `block->bbNext` higher up that we can capture too.

Reply:
Good idea, I'll push a revision